packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:io';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/auth_logger_mixin.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/shorebird_build_mixin.dart';
+import 'package:shorebird_cli/src/shorebird_config_mixin.dart';
+import 'package:shorebird_cli/src/shorebird_validation_mixin.dart';
+
+/// {@template build_ipa_command}
+/// `shorebird build ipa`
+/// Builds an .xcarchive and optionally .ipa for an iOS app to be generated for
+/// App Store submission.
+/// {@endtemplate}
+class BuildIpaCommand extends ShorebirdCommand
+    with
+        AuthLoggerMixin,
+        ShorebirdValidationMixin,
+        ShorebirdConfigMixin,
+        ShorebirdBuildMixin {
+  /// {@macro build_ipa_command}
+  BuildIpaCommand({required super.logger, super.auth, super.validators}) {
+    argParser
+      ..addOption(
+        'target',
+        abbr: 't',
+        help: 'The main entrypoint file of the application.',
+      )
+      ..addOption(
+        'flavor',
+        help: 'The product flavor to use when building the app.',
+      )
+      ..addFlag(
+        'codesign',
+        help:
+            '''Codesign the application bundle (only available on device builds).''',
+      );
+  }
+
+  @override
+  String get description =>
+      '''Builds an .xcarchive and optionally .ipa for an iOS app to be generated for App Store submission.''';
+
+  @override
+  String get name => 'ipa';
+
+  @override
+  Future<int> run() async {
```

Comment:
Do we want to verify that the current OS can build ipas? I'm sure `flutter build ipa` will spit up appropriately so maybe we don't really need to.